web/src/test/kotlin/no/nav/su/se/bakover/web/TestEnvironment.kt

```diff
@@ -147,55 +148,61 @@ internal fun Application.testSusebakover(
-): TestApplicationCall {
-    return handleRequest(method, uri) {
-        addHeader(HttpHeaders.XCorrelationId, DEFAULT_CALL_ID)
-        addHeader(HttpHeaders.Authorization, jwtStub.createJwtToken(roller = roller).asBearerToken())
+    roller: List<Brukerrolle> = emptyList(),
+    setup: HttpRequestBuilder.() -> Unit = {},
+): HttpResponse {
+    return this.client.request(uri) {
+        this.method = method
+        this.headers {
+            append(HttpHeaders.XCorrelationId, DEFAULT_CALL_ID)
+            append(HttpHeaders.Authorization, jwtStub.createJwtToken(roller = roller).asBearerToken())
+        }
         setup()
     }
 }
 
-fun TestApplicationEngine.defaultRequest(
+suspend fun ApplicationTestBuilder.defaultRequest(
     method: HttpMethod,
     uri: String,
-    roller: List<Brukerrolle>,
+    roller: List<Brukerrolle> = emptyList(),
     navIdent: String,
-    setup: TestApplicationRequest.() -> Unit = {},
-): TestApplicationCall {
-    return handleRequest(method, uri) {
-        addHeader(HttpHeaders.XCorrelationId, DEFAULT_CALL_ID)
-        addHeader(
-            HttpHeaders.Authorization,
-            jwtStub.createJwtToken(roller = roller, navIdent = navIdent).asBearerToken(),
-        )
+    setup: HttpRequestBuilder.() -> Unit = {},
+): HttpResponse {
+    return this.client.request(uri) {
+        this.method = method
+        this.headers {
+            append(HttpHeaders.XCorrelationId, DEFAULT_CALL_ID)
+            append(HttpHeaders.Authorization, jwtStub.createJwtToken(roller = roller, navIdent = navIdent).asBearerToken())
+        }
         setup()
     }
 }
 
-fun TestApplicationEngine.requestSomAttestant(
+suspend fun ApplicationTestBuilder.requestSomAttestant(
     method: HttpMethod,
     uri: String,
     navIdent: String? = null,
-    setup: TestApplicationRequest.() -> Unit = {},
-): TestApplicationCall {
-    return handleRequest(method, uri) {
-        addHeader(HttpHeaders.XCorrelationId, DEFAULT_CALL_ID)
-        addHeader(
-            HttpHeaders.Authorization,
-            jwtStub.createJwtToken(roller = listOf(Brukerrolle.Attestant), navIdent = navIdent).asBearerToken(),
-        )
+    setup: HttpRequestBuilder.() -> Unit = {},
+): HttpResponse {
+    return this.client.request(uri) {
+        this.method = method
+        this.headers {
+            append(HttpHeaders.XCorrelationId, DEFAULT_CALL_ID)
+            append(
+                HttpHeaders.Authorization,
+                jwtStub.createJwtToken(roller = listOf(Brukerrolle.Attestant), navIdent = navIdent).asBearerToken(),
+            )
+        }
         setup()
     }
 }
 
-fun TestApplicationEngine.requestSomAttestant(
+suspend fun ApplicationTestBuilder.requestSomAttestant(
     method: HttpMethod,
     uri: String,
-): TestApplicationCall {
-    return requestSomAttestant(method, uri, null) {}
+): HttpResponse {
+    return requestSomAttestant(method = method, uri = uri, navIdent = null) {}
```

Comment:
Usikker på om null er en god default her, dette bør fele for alle kall som kommer forbi auth.